internal/daemon/common/client_ip.go

```diff
@@ -34,6 +34,12 @@ func ClientIpFromRequest(ctx context.Context, listenerCfg *listenerutil.Listener
 		if err != nil {
 			return "", errors.Wrap(ctx, err, op)
 		}
+		if listenerCfg.Type == "unix" {
+			// Some platforms (Linux) use "@" in this case but some like Mac
+			// leave it empty which causes issues with the rate limiting logic,
+			// so standardize on "@" in this case.
+			return "@", nil
+		}
```

Comment:
Should we only do this if `ip` is `""`? ie:

```golang
if listenerCfg.Type == "unix" && ip == "" {
```

It probably results in the same thing, but it would prevent us from changing the value if for some reason it gets set to something other than `@` by some platform either now or in the future, not that it should really matter much.

Reply:
Sure, will change.